### PR TITLE
Fix in-flight human intervention loop

### PR DESCRIPTION
**Pull request by subsetpark**
## Summary

- keep the Human intervention exemption active after guidance moves from the queue to in-flight delivery
- prevent capped no-PR Human starts from immediately becoming stale and force-completing in a loop
- add a regression covering repeated no-commit Human starts and the next accepted attempt

## Validation

- `dune build`
- `dune runtest`
- `dune fmt`
- `git diff --check`

---
<a href="https://app.blacksmith.sh/flowglad/codesmith/onton/pr/432"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792258024&installation_model_id=19519&pr_number=432&repository=flowglad%2Fonton&return_to=https%3A%2F%2Fgithub.com%2Fflowglad%2Fonton%2Fpull%2F432&signature=e2dc0d9f711415a3952779487873dac99df3e317318c3144a1827199e3d17754"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

- **Bug Fixes**
  - Human guidance now remains active while an accepted action is being delivered, preventing it from becoming stale due to failure limits.
  - Agents remain runnable after repeated failed Human-start attempts instead of incorrectly entering intervention.
  - Intervention status now clears correctly after Human guidance is accepted.

### CODE REVIEW

File: lib_core/patch_agent.mli [hunk 1]
```diff
@@ -248,7 +248,8 @@ val needs_intervention : t -> bool
     - [is_pr_missing t] (PR vanished from the remote — bypasses the Human
       exemption; queued Human entries are deferred until [Missing → Present]
       recovery rather than dispatched while [Missing])
-    - [Human] not in queue AND any of: [ci_failure_count >= max_ci_failures],
+    - no queued or in-flight [Human] delivery AND any of:
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib_core/patch_agent.mli:252]: Fixed. The raw intervention APIs now take `human_pending`, explicitly representing either queued or in-flight Human delivery. `Activity_log_sink` derives that value from both the persisted `queue` and `inflight_human_messages`, so telemetry now matches the live-agent exemption after ownership transfers to the runner. I also added an inline regression test covering an empty queue, a persisted in-flight Human message, and a CI count at the intervention threshold. `dune build` and `dune runtest` pass.
